code/game/machinery/doors/airlock_types.dm

```diff
@@ -504,19 +508,40 @@
 
 /obj/machinery/door/airlock/cult/allowed(mob/living/L)
 	if(!density)
-		return 1
+		return TRUE
 	if(friendly || iscultist(L) || isshade(L)|| isconstruct(L))
-		new openingoverlaytype(loc)
-		return 1
+		if(!stealthy)
+			new openingoverlaytype(loc)
+		return TRUE
 	else
-		new /obj/effect/temp_visual/cult/sac(loc)
-		var/atom/throwtarget
-		throwtarget = get_edge_target_turf(src, get_dir(src, get_step_away(L, src)))
-		L << pick(sound('sound/hallucinations/turn_around1.ogg',0,1,50), sound('sound/hallucinations/turn_around2.ogg',0,1,50))
-		L.Weaken(2)
-		spawn(0)
+		if(!stealthy)
+			new /obj/effect/temp_visual/cult/sac(loc)
+			var/atom/throwtarget
+			throwtarget = get_edge_target_turf(src, get_dir(src, get_step_away(L, src)))
+			L << pick(sound('sound/hallucinations/turn_around1.ogg',0,1,50), sound('sound/hallucinations/turn_around2.ogg',0,1,50))
+			L.Weaken(2)
+			spawn(0)
```

Comment:
Why the spawn(0)?

Reply:
no clue, was from previous cult airlock code, i removed it and it works fine